### ISSUE TITLE
Unable to change Compute Offering, if the identical Disk Offering is linked to it

### ISSUE THREAD

**Issue by s-seitz** [apache/cloudstack#6679]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME

~~~
 API
~~~

##### CLOUDSTACK VERSION

~~~
4.17.0.1-shapeblue0
~~~

##### OS / ENVIRONMENT

Ubuntu 20.04.5 LTS, obviously not relevant

##### SUMMARY

If different Compute offerings have the identical Disk offering linked to, it is impossible to change ("Scale") the particular offerings for a VM.
Instead following error is raised: "Volume [UUID] already have the new disk offering [UUID] provided."

##### STEPS TO REPRODUCE

~~~
* Create a unique Disk offering
* Create at least two dfferent Compute offerings, linking the formerly created Disk offering to both of them.
* Create a VM with one of the formerly created Compute offerings
* Power down the VM (in our case we do not use dynamic scalability)
* Scale the VM to use the other created Compute offering.
~~~

##### EXPECTED RESULTS

~~~
No error, just change to the values of the given Compute Offering without touching the Disk, as there's no need to since the Disk Offering stays the same.
~~~

##### ACTUAL RESULTS

~~~
An error is raised "Volume [UUID] already have the new disk offering [UUID] provided.
~~~

**Comment by shwstppr** [comment]
@s-seitz Thank you for reporting the issue. I've created PR #6688 to fix it cc @harikrishna-patnala

**Comment by shwstppr** [comment]
Fixed with #6688
